deepchem/molnet/load_function/molnet_loader.py

```diff
@@ -0,0 +1,139 @@
+"""
+Common code for loading MoleculeNet datasets.
+"""
+import os
+import logging
+import deepchem as dc
+from deepchem.data import Dataset, DiskDataset
+from typing import List, Optional, Tuple, Union
+
+logger = logging.getLogger(__name__)
+
+featurizers = {
+    'ecfp': dc.feat.CircularFingerprint(size=1024),
+    'graphconv': dc.feat.ConvMolFeaturizer(),
+    'weave': dc.feat.WeaveFeaturizer(),
+    'raw': dc.feat.RawFeaturizer(),
+    'smiles2img': dc.feat.SmilesToImage(img_size=80, img_spec='std')
+}
+
+splitters = {
+    'index': dc.splits.IndexSplitter(),
+    'random': dc.splits.RandomSplitter(),
+    'scaffold': dc.splits.ScaffoldSplitter(),
+    'butina': dc.splits.ButinaSplitter(),
+    'task': dc.splits.TaskSplitter(),
+    'stratified': dc.splits.RandomStratifiedSplitter()
+}
+
+
+class _MolnetLoader(object):
+  """The class provides common functionality used by many molnet loader functions.
+  It is an abstract class.  Subclasses implement loading of particular datasets.
+  """
+
+  def __init__(self, featurizer: Union[dc.feat.Featurizer, str],
+               splitter: Union[dc.splits.Splitter, str, None],
+               data_dir: Optional[str], save_dir: Optional[str], **kwargs):
```

Comment:
Perhaps we can add a `transformers` kwarg here to allow for user specified transformations.

Reply:
We can't pass in transformers, because some transformers require the dataset as a constructor argument.  That means the transformers can't be created until after the dataset has already been loaded.

Reply:
Good point ! As a simple possibility, perhaps we could allow for a `no_transformers` option so users can obtain the dataset without any transformations applied? They can then apply their own transformations as desired